src/rtctools_heat_network/workflows/io/write_output.py

```diff
@@ -1026,10 +1039,17 @@ def _name_to_asset(name):
                     #     tags=optim_simulation_tag,
                     # )
                     # ------------------------------------------------------------------------------
-
-                except Exception:
+                except StopIteration:
                     # If the asset has been deleted, thus also not placed
                     pass
+                except Exception:  # TODO fix other places in the where try/except end with pass
+                    logger.error(
```

Comment:
Tip: If you use `.exception` instead of `.error` the logging framework will print out the stacktrace without you needed to call `traceback.print_exc()`

Question: Why do you still catch Exception here? To add the debug info? The logic right now is to continue the program after catching the exception. Perhaps you would rather the program stop after an Exception happens? If so, you can just not catch the Exception or reraise it with `raise`. You know what fits best of course but I want to provide you with some options to choose form.